regions/core/core.py

```diff
@@ -293,6 +303,205 @@ def to_pixel(self, wcs):
         raise NotImplementedError
 
 
+@six.add_metaclass(abc.ABCMeta)
+class RegionAttr(object):
+
+    def __init__(self):
+        self._values = weakref.WeakKeyDictionary()
+
+    def __get__(self, instance, owner):
+        if instance is None:
+            return self
+        return self._values.get(instance, None)
+
+    def __set__(self, instance, value):
+        self._validate(value)
+        self._values[instance] = value
+
+    def _validate(self, value):
+        raise NotImplementedError
+
+
+class CenterPix(RegionAttr):
+
+    def _validate(self, value):
+        if isinstance(value, PixCoord) and value.isscalar:
+            return
+        else:
+            raise ValueError('The center must be a 0D PixCoord object')
```

Comment:
might this be simpler as an `if not(condition): raise`, skipping the return statement above?